scripts/update-command.py

```diff
@@ -0,0 +1,286 @@
+    def removesuffix(self, suffix: str) -> str:
+        return self[: -len(suffix)] if suffix and self.endswith(suffix) else self
+
+
+class MyFormatter(logging.Formatter):
+    grey = "\x1b[0;30m"
+    yellow = "\x1b[33;20m"
+    red = "\x1b[31;20m"
+    bold_red = "\x1b[31;1m"
+    reset = "\x1b[0m"
+    format = "%(levelname)s: %(message)s (%(filename)s:%(lineno)d)"
+
+    FORMATS = {
+        logging.INFO: grey + format + reset,
+        logging.WARNING: yellow + format + reset,
+        logging.ERROR: red + format + reset,
+    }
+
+    def format(self, record):
+        log_fmt = self.FORMATS.get(record.levelno)
+        formatter = logging.Formatter(log_fmt)
+        return formatter.format(record)
+
+
+logger = logging.getLogger(__name__)
+logger.propagate = False
+
+ch = logging.StreamHandler()
+ch.setFormatter(MyFormatter())
+
+logger.addHandler(ch)
+
+
+def get_locales(base_path: Path) -> list[str]:
+    return [
+        d.name.split(".")[1]
+        for d in base_path.iterdir()
+        if d.is_dir() and d.name.startswith("pages.")
+    ]
+
+
+def get_tldr_pages(platform_path: Path) -> list[Path]:
+    return [f for f in platform_path.iterdir() if f.is_file() and f.suffix == ".md"]
+
+
+def take_command_with_common_part(commands: list[str], common_part: str) -> str:
+    return next(
+        (
+            f"`{command}`"
+            for command in commands
+            if remove_placeholders(command) == common_part
+        ),
+        None,
+    )
+
+
+def get_commands_of_page(page_text: str) -> list[str]:
+    command_pattern = re.compile(r"`([^`]+)`")
+    return re.findall(command_pattern, page_text)
+
+
+def find_command_with_common_part(common_part: str, page_text: str) -> list[str]:
+    commands = get_commands_of_page(page_text)
+    return take_command_with_common_part(commands, common_part)
+
+
+def get_page_path(tldr_root: Path, locale: str, platform: str, filename: str):
+    if locale == "":
+        return tldr_root / "pages" / platform / filename
+    return tldr_root / f"pages.{locale}" / platform / filename
+
+
+def split_by_curly_brackets(s: str) -> list[str]:
+    return re.split(r"(\{\{.*?\}\})", s)
+
+
+def parse_placeholders(command: str) -> list[str]:
+    return [
+        part.strip("{}")
+        for part in split_by_curly_brackets(command)
+        if part.startswith("{{") and part.endswith("}}")
+    ]
+
+
+def is_placeholder(s: str) -> bool:
+    return s.startswith("{{") and s.endswith("}}")
+
+
+def place_placeholders(command: str, placeholders: list[str]) -> str:
+    return reduce(
+        lambda cmd, ph: cmd.replace("{{}}", "{{" + ph + "}}", 1), placeholders, command
+    )
+
+
+def remove_placeholders(command: str) -> str:
+    return re.sub(r"\{\{.*?\}\}", "{{}}", command)
+
+
+def add_backticks(command: str) -> str:
+    return "`" + command.strip("`") + "`"
+
+
+def update_page(
+    page_path: Path,
+    old_common_part: str,
+    new_common_part: str,
+    place_placeholders_function,
```

Comment:
I think `place_placeholders` can be directly called rather than passed as `place_placeholders_function` argument. Is there any reason to do so?